extension/deployment/src/main/java/io/quarkiverse/infinispan/embedded/deployment/InfinispanEmbeddedProcessor.java

```diff
@@ -85,6 +96,25 @@ void setup(BuildProducer<ReflectiveClassBuildItem> reflectiveClass,
             }
         }
 
+        // Protostream
+        Index index = applicationIndexBuildItem.getIndex();
+        Collection<ClassInfo> initializerClasses = index.getAllKnownImplementors(DotName.createSimple(
+                SerializationContextInitializer.class.getName()));
+        initializerClasses
+                .addAll(index.getAllKnownImplementors(DotName.createSimple(GeneratedSchema.class.getName())));
+        Set<String> initializers = new HashSet<>(initializerClasses.size());
+        for (ClassInfo ci : initializerClasses) {
```

Comment:
Why do we need to load the class here? I'm not sure this will always work depending on the classpath (I could be wrong).

Can we not use `public ServiceProviderBuildItem(String serviceInterfaceClassName, List<String> providers)` which does not require a Class object?

Reply:
@ryanemerson This is the way it is done in the infinispan client to load the serialization context. I for inspired by that and the other examples provided. I don't know if what you mentions works, I will try